### ISSUE TITLE
Training with local variables is very slow in distributed tf

### ISSUE THREAD

**Issue by hellolovetiger** [tensorflow/tensorflow#6475]
I used local variables on worker tasks in a distributed RNN training. Every update go through 3 steps: 1) assign local variables on each worker with PS variables 2) training local variables in several loops/batches, and accumulate the gradients from each loop 3) apply the accumulated gradients to the PS variables. The first and second step is very slow: for step 1, it takes about 12 seconds; for step 2, every loop takes about 6 seconds. If I combine the three steps together (not use local variable and go through one batch data every update), it only takes about 0.7 second. So, I suspect that it is due to the bad performance of local variables. 

### Environment info
Ubuntu 14.04, CUDA 8.0, tensorflow 0.12.0-rc1

**Comment by yaroslavvb** [comment]
How big are your local variables? There's some slowness in current version if they are too big -- https://github.com/tensorflow/tensorflow/issues/6116 . Also, what does your `session.run` looks like? If you do something like `sess.run(assign_op)` then you are fetching result to Python which can be slow (https://github.com/tensorflow/tensorflow/issues/6256), the solution is to do `sess.run(assign_op.op)`

**Comment by hellolovetiger** [comment]
Yes, my variables are very big, totally larger than 400MB. `sess.run(assign_op.op)` solved my problem. Thanks a lot.
